dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -245,4 +245,33 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
             LOG_IF_FAILED(::CoRevokeClassObject(static_cast<DWORD>(token.Cookie())));
         }
     }
+
+    bool PushNotificationManager::IsBackgroundTaskBuilderAvailable()
+    {
+        auto builder{ BackgroundTaskBuilder() };
+        auto builder5{ builder.try_as<winrt::IBackgroundTaskBuilder5>() };
+        return builder5 != nullptr;
+    }
+
+    bool PushNotificationManager::IsActivatorSupported(PushNotificationRegistrationOptions const& options)
+    {
+        THROW_HR_IF(E_INVALIDARG, options == PushNotificationRegistrationOptions::Undefined); // WI_IsFlagSet does not allow 0 as flag.
+
+        auto isBackgroundTaskFlagSet{ WI_IsAnyFlagSet(options, PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator) };
+        auto isProtocolActivatorSet{ WI_IsFlagSet(options, PushNotificationRegistrationOptions::ProtocolActivator) };
+
+        THROW_HR_IF(E_INVALIDARG, isBackgroundTaskFlagSet && isProtocolActivatorSet); // Invalid flag combination
+        if (AppModel::Identity::IsPackagedProcess() && IsBackgroundTaskBuilderAvailable())
+        {
+            if (isProtocolActivatorSet)
```

Comment:
Please add a comment here that we won't support protocol activation if COM activation is supported in the flow.

That mean's @sharath2727 should be calling this API in his channel registration flow as well rather than adding his own custom logic.